src/tools/prefs.mac.mm

```diff
@@ -70,6 +98,58 @@ void clear_pref(std::string keypath) {
 }
 
 bool sync_prefs() {
+	if(recording){
+		std::ostringstream prefs_recording;
+		NSUserDefaults* standard = [NSUserDefaults standardUserDefaults];
+		for(id pref in prefsToRecord) {
+			id object = [standard objectForKey: pref];
+			if(object != nil){
+				prefs_recording << [pref UTF8String] << " = ";
+				NSInteger type = [prefsToRecord[pref] integerValue];
+				bool bvalue;
```

Comment:
I think I'd prefer to declare these in the switch statement with the following idiom:

```cpp
switch(int(type)) {
  case kBool: {
    bool bvalue
    // ...
  } break;
}
```